frontend/my-adopthive-app/src/AdopterFavourites/AdopterFavourites.jsx

```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect, useContext } from 'react';
+import { UserContext } from '../UserContext';
+import './AdopterFavourites.css';
+
+const AdopterFavourites = () => {
+  const { user } = useContext(UserContext);
+  const [favourites, setFavourites] = useState([]);
+  const [error, setError] = useState('');
+  const [preferences, setPreferences] = useState({});
+
+  useEffect(() => {
+    const fetchFavourites = async () => {
+      if (!user || !user.id) {
+        setError('User not logged in');
+        return;
+      }
+      try {
+        const response = await fetch(`http://localhost:3004/favourites?userId=${user.id}`);
+        if (!response.ok) {
+          throw new Error('Failed to fetch favourites');
+        }
+        const data = await response.json();
+        setFavourites(data);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+    fetchFavourites();
```

Comment:
I see this logic is duplicated between `AdopterFavorites` and `AdopteeFavorites`. Can you try to create a Util to consolidate the logic, and reference that in each class?

Reply:
I have removed the rank which makes both duplicated as it is not needed in favourites